src/stream/src/executor/chain.rs

```diff
@@ -184,11 +184,19 @@ mod test {
         let chain = ChainExecutor::new(first, second, vec![0], notifier, schema);
 
         let mut chain = Box::new(chain).execute();
+        chain
+            .next()
+            .await
+            .transpose()
+            .unwrap()
+            .unwrap()
+            .as_barrier()
+            .expect("the first message received by chain must be a barrier");
 
-        let mut count = 0;
+        let mut count = 3;
         while let Some(Message::Chunk(ck)) = chain.next().await.transpose().unwrap() {
-            count += 1;
             assert_eq!(ck, StreamChunk::from_pretty(&format!("I\n + {count}")));
+            count += 1;
         }
```

Comment:
Could you please also test the `count` here to check how many chunks we received?